front50-s3/src/main/java/com/netflix/spinnaker/front50/plugins/S3PluginBinaryStorageService.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.front50.plugins;
+
+import static java.lang.String.format;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.*;
+import com.google.common.hash.Hashing;
+import com.google.common.io.ByteStreams;
+import com.netflix.spinnaker.front50.config.S3Properties;
+import com.netflix.spinnaker.kork.exceptions.SystemException;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class S3PluginBinaryStorageService implements PluginBinaryStorageService {
+
+  private final AmazonS3 amazonS3;
+  private final S3Properties properties;
+
+  public S3PluginBinaryStorageService(AmazonS3 amazonS3, S3Properties properties) {
+    this.amazonS3 = amazonS3;
+    this.properties = properties;
+  }
+
+  @Override
+  public void store(@Nonnull String key, @Nonnull byte[] item) {
+    ObjectMetadata metadata = new ObjectMetadata();
+    metadata.setContentLength(item.length);
+    metadata.setContentMD5(Hashing.md5().hashBytes(item).toString());
+
+    amazonS3.putObject(
```

Comment:
What happens here if the binary exists?

Reply:
Oh, it throws an exception. I'm surprised you didn't see this. 🤐